services/web-server/src/loaders/tasks.js

```diff
@@ -3,16 +3,15 @@ const sift = require('../utils/sift');
 const fetch = require('../utils/fetch');
 const ConnectionLoader = require('../ConnectionLoader');
 const Task = require('../entities/Task');
+const tryCatch = require('../utils/tryCatch');
 
 module.exports = ({ queue, index }) => {
   const task = new DataLoader(taskIds =>
-    Promise.all(
-      taskIds.map(async taskId => {
-        const task = await queue.task(taskId);
+    Promise.all(taskIds.map(async (taskId) => {
+      const [error, task] = await tryCatch(queue.task(taskId));
 
-        return new Task(taskId, null, task);
-      }),
-    ),
+      return error || new Task(taskId, null, task);
```

Comment:
Will returning an Error object do the right thing, or would this need to be thrown?

Reply:
> Will returning an Error object do the right thing

Good question. The dataloader package seems to check for the instance of the object so we are good.

https://github.com/graphql/dataloader/blob/efc2b1a376863daeee88ff8e3bc901bb54fc5356/src/index.js#L293

Reply:
Good question. The dataloader package seems to check for the instance of the object so we are good.

https://github.com/graphql/dataloader/blob/efc2b1a376863daeee88ff8e3bc901bb54fc5356/src/index.js#L293